modules/host/contractmanager/sectorupdate.go

```diff
@@ -551,17 +551,22 @@ func (cm *ContractManager) RemoveSectorBatch(sectorRoots []crypto.Hash) error {
 	var wg sync.WaitGroup
 	// Ensure only 'maxSectorBatchThreads' goroutines are running at a time.
 	semaphore := make(chan struct{}, maxSectorBatchThreads)
+	i := 0
 	for _, root := range sectorRoots {
 		wg.Add(1)
 		semaphore <- struct{}{}
-		go func(root crypto.Hash) {
+		i++
+		go func(root crypto.Hash, i int) {
 			id := cm.managedSectorID(root)
 			cm.wal.managedLockSector(id)
 			cm.wal.managedRemoveSector(id) // Error is ignored.
 			cm.wal.managedUnlockSector(id)
+			if i%1000 == 0 {
+				cm.log.Printf("Finished %v sectors\n", i)
```

Comment:
I could change logging instead of producing a line every 1k sectors, log every 5 min. Thus throughput of the log records flow will be fixed.

Reply:
My main concern is that the information has no context associated with it, just that sectors are being removed. It's still missing "why", but I think a better route to take for this information would be to add to the alerter:
```go
func (cm *ContractManager) RemoveSectorBatch(sectorRoots []crypto.Hash) error {
	// Prevent shutdown until this function completes.
	err := cm.tg.Add()
	if err != nil {
		return err
	}
	defer cm.tg.Done()

	// unique alert id for each call
	alertID := modules.AlertID("cm-removing-sectors-" + hex.EncodeToString(fastrand.Bytes(12)))
	start := time.Now()
	// helper to format the alert.
	updateRemovalAlert := func(i, n int) {
		cm.staticAlerter.RegisterAlert(alertID, fmt.Sprintf("Removed %v/%v sectors in %v", i, n, time.Since(start)), "", modules.SeverityWarning)
	}
	// unregister the alert.
	defer cm.staticAlerter.UnregisterAlert(alertID)

	// Add each sector in a separate goroutine.
	var wg sync.WaitGroup
	// Ensure only 'maxSectorBatchThreads' goroutines are running at a time.
	semaphore := make(chan struct{}, maxSectorBatchThreads)
	i, n := 0, len(sectorRoots)
	for _, root := range sectorRoots {
		wg.Add(1)
		semaphore <- struct{}{}
		i++
		updateRemovalAlert(i, n)
		go func(root crypto.Hash) {
			id := cm.managedSectorID(root)
			cm.wal.managedLockSector(id)
			cm.wal.managedRemoveSector(id) // Error is ignored.
			cm.wal.managedUnlockSector(id)
			<-semaphore
			wg.Done()
		}(root)
	}
	wg.Wait()
	return nil
}
```